packages/gitbook/src/lib/api.ts

```diff
@@ -381,10 +394,16 @@ const getAPIContextId = async () => {
  * Get a revision by its ID.
  */
 export const getRevision = cache({
-    name: 'api.getRevision.v2',
-    tag: (spaceId, revisionId) =>
-        getCacheTag({ tag: 'revision', space: spaceId, revision: revisionId }),
-    tagImmutable: true,
+    name: 'api.getRevision.v3',
+    tag: (spaceId, revisionId, fetchOptions) =>
+        // Temporary hack to make it work with OpenAPI on v1
+        fetchOptions.tags?.[0] ??
+        getCacheTag({
+            tag: 'revision',
+            space: spaceId,
+            revision: revisionId,
+        }),
+    tagImmutable: false,
```

Comment:
Should we create 2 cache function to avoid churning all existing cache and keeping `tagImmutable`?

I'm slightly worried about deploying such a changes and getting 100k requests to our API at once.

Basically I think `getRevision` should stay the same cache parameters and we pass `computed: false`, and then we have the other function for the second case?

Reply:
I added them, but in v1 we will switch to the other function, so all v1 will be refetched because if the result is computed we need the tag (currently we don't have them).